contributors/design-proposals/local-storage-pv.md

```diff
@@ -0,0 +1,631 @@
+# Local Storage Persistent Volumes
+
+Authors: @msau42, @vishh
+
+This document presents a detailed design for supporting persistent local storage,
+as outlined in [Local Storage Overview](local-storage-overview.md).
+Supporting all the use cases for persistent local storage will take many releases,
+so this document will be extended for each new release as we add more features.
+
+## Goals
+
+* Allow pods to mount any local block or filesystem based volume.
+* Allow pods to mount dedicated local disks, or channeled partitions as volumes for
+IOPS isolation.
+* Allow pods do access local volumes without root privileges.
+* Allow pods to access local volumes without needing to undestand the storage
+layout on every node.
+* Persist local volumes and provide data gravity for pods.  Any pod
+using the local volume will be scheduled to the same node that the local volume
+is on.
+* Allow tolerations on volumes to be specified to release the volume during
```

Comment:
not sure i understand this. can we explain this more clear?

Reply:
The default behavior with local storage is that if the node goes down, your pod cannot be rescheduled because you're bound to that local volume.  If your application uses replication, it can potentially be rescheduled, on a brand new node, with a new volume and then resync the replica.  So the idea of tolerations here is similar to the pod tolerations feature, except at a volume level.  The tolerations setting will determine how long we wait on specific error conditions (like node unavailable) before we unbind the PVC to find a new one.

Reply:
ok. makes sense. basically pod just wants a local pv, but can forgive its state. maybe we can make this clear in the doc.

Reply:
Agree, I tried to reword it to make it sound scarier.